embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/clock/Clock.kt

```diff
@@ -9,4 +10,15 @@ public fun interface Clock {
      * Returns the current milliseconds from epoch.
      */
     public fun now(): Long
+
+    /**
+     * Returns the current nanoseconds from epoch
+     */
+    public fun nowInNanos(): Long = now().millisToNanos()
```

Comment:
```suggestion
    public fun now(unit: TimeUnit = TimeUnit.MILLISECONDS): Long
```

I have a slight preference for this syntax as it avoids the proliferation of other functions at a later date such as millisToSeconds etc. Feel free to merge as-is if you disagree.

Reply:
is the net effect of this that we add 6 zeros to the end of the time value?

Reply:
> is the net effect of this that we add 6 zeros to the end of the time value?

Basically. Are you concerned about perf? If so I can write it in a way that makes the compiler inline this for us. But it's not going to make a huge difference since we don't call this that often.

Reply:
no, it's fine. i just wanted to understand if i was missing something in terms of precision. i believe the other clock options we may use in the future, [this](https://developer.android.com/reference/android/os/SystemClock#currentGnssTimeClock()) and [this](https://developer.android.com/reference/android/os/SystemClock#currentNetworkTimeClock()), also seem to just have ms precision. the only thing that seems to have ns precision -- not to be confused with ns accuracy :) -- is [this](https://developer.android.com/reference/android/location/GnssClock#getTimeNanos())

Reply:
I don't know if the Android clock is even capable of that level of precision. It's certainly not useful for us. Microseconds I can get behind, but this is a bit of an overkill for our use case. Alas, that's the spec.